cql3/functions/functions.cc

```diff
@@ -455,7 +455,7 @@ function_call::bind_and_get(const query_options& options) {
         buffers.push_back(std::move(to_bytes_opt(val)));
     }
     auto result = execute_internal(options.get_cql_serialization_format(), *_fun, std::move(buffers));
-    return options.make_temporary(cql3::raw_value::make_value(result));
+    return cql3::raw_value_view::make_temporary(cql3::raw_value::make_value(result));
 }
```

Comment:
This fixes another weakness of the current system - a long running query can cause a temporary leak of temporaries. The leak is plugged when the query terminates, but if the query runs for a long time, the reduction in available memory can be bad (or cause OOM).

Reply:
But is a long running query really capable of leaking a considerable amount of temporaries? I was only able to see these temporaries instantiated once for some cql constants, and these are usually only a few (dozens) bytes long - would it even be noticeable overhead, given that long queries usually also take lots of other resources, like cache and readers? (I'm not trying to undermine the sense of my own patch, just curious)

Reply:
Consider

    SELECT count(my_func(foo)) FROM tab;

Every iteration will generate a temporary for the result of my_func. Before your patch, those temporaries would accumulate in query_options::_temporaries. The query would quickly time out and be abandoned by the user, but would still continue executing and pushing a word into temporary storage until tab was exhausted, or memory, or the remaining entropy reserve of the universe (whichever comes first).

Reply:
I'm not happy with the transformation of raw_value_view into a heavyweight object, but the situation now is much better than before.